tests/suites/test_suite_mpi.function

```diff
@@ -1,5 +1,36 @@
 /* BEGIN_HEADER */
 #include "mbedtls/bignum.h"
+
+typedef struct mbedtls_test_mpi_random
+{
+    data_t *data;
+    size_t  pos;
+    size_t  chunk_len;
+} mbedtls_test_mpi_random;
+
+/*
+ * This function is called by the mbedtls_mpi_bad_trng_check_test in order to
+ * simluate a faulty TRNG.
+*/
+int mbedtls_test_random_determinizer_fixed( void* state, unsigned char* buf,
+                                      size_t len )
+{
+    mbedtls_test_mpi_random *random = (mbedtls_test_mpi_random*) state;
+
+    if( random == NULL || random->data->x == NULL || buf == NULL )
+        return -1;
+
+    if( random->data->len < len )
+    {
+        return -1;
+    }
+
+    memset( buf, 0, len );
+    memcpy( buf , &random->data->x, len );
```

Comment:
Remove `&` on `random->data->x`. Found via ASan which detected the out-of-bounds access on `random->data`.